api/queries_pr.go

```diff
@@ -590,7 +590,21 @@ loop:
 		}
 	}
 
-	return prs, nil
+	return removeDuplicates(prs), nil
```

Comment:
This will remove duplicates, which may return fewer results than the user has requested.

Let's say the user requests `gh pr list --limit 10`. The user will be expecting 10 results if there are at least 10 open PRs in the repository, and if any fewer results are returned then this will be perceived as a bug.

So instead of removing duplicates here, could you perhaps make an attempt to remove them at the pagination level? https://github.com/cli/cli/blob/d4cb2d860f03631c24a74717372356986eec6327/api/queries_pr.go#L578-L583

Basically, that would mean that you would skip the `append` if the item already appears in the collection so far. Let me know if that makes sense!

Reply:
Will do.